### PR TITLE
Add Makefile and draft of README

### PR DESCRIPTION
**Pull request by yimjay**
Address AA-11, AA-7

Summary:
- Add Makefile
- Add initial draft of README file specifying requirements, steps, and contribution guidelines
- .gitattributes file to normalize linebreaks between contributors to use LF instead of CRLF

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+input=SinglyLinkedList.h_orig
+output=
+debug=
+POET_HOME=../../../POET/trunk
+POET=pcg
+
+now:
```
Comment (qingyi-yan) [inline Makefile:7]: Change the "now" rule to "all", if you mean to analyze all included abstractions against a given input?

File: README.md [hunk 2]
```diff
@@ -1 +1,38 @@
-<*Owners: *>
+
+## About the project
+
+[Brief explanation of goal of project]
```
Comment (qingyi-yan) [inline README.md:4]: This should be replaced with the actual content, something like "This project aims to automatically discover the higher-level semantics of C++ classes by classifying them as abstractions such as stacks and queues"?

File: README.md [hunk 3]
```diff
@@ -1 +1,38 @@
-<*Owners: *>
```
Comment (qingyi-yan) [inline README.md:1]: Replace with a list of the owner handles?

File: README.md [hunk 4]
```diff
@@ -1 +1,38 @@
...
+
+## About the project
+
+[Brief explanation of goal of project]
+
+### Current status
+
+#### First prototype
+
+[Brief description of current status]
```
Comment (qingyi-yan) [inline README.md:10]: Replace with actual content?

File: README.md [hunk 5]
```diff
@@ -1 +1,38 @@
...
+
+### Repository
+
+- `research/abstractionAnalysis/` directory contains all implementations and necessary files for this project
+- `research/abstractionAnalysis/testfiles` directory contains test input files and expected outputs
+- `research/abstractionAnalysis/semantics` directory contains semantic specification files used by the parser
+- `research/abstractionAnalysis/design` directory contains initial prototype design files (should it be included in public repository?)
```
Comment (qingyi-yan) [inline README.md:17]: Do not put this in the public release repo.

File: README.md [hunk 6]
```diff
@@ -1 +1,38 @@
...
+
+## Steps
+
```
Comment (qingyi-yan) [inline README.md:20]: Add steps on how to combine this repo and the POET repo?

File: .gitattributes [hunk 7]
```diff
@@ -0,0 +1 @@
+* text eol=lf
```
Comment (bnestere) [inline .gitattributes:1]: Ah now I understand the previous Windows line endings issue you had been talking about. Good catch with using `.gitattributes`! As you're adding it into the repo, can you make it more complete, e.g. by following https://rehansaeed.com/gitattributes-best-practices/ ?

File: Makefile (hunk 1 above)
Comment (yimjay) [reply Makefile]: Updated

File: README.md (hunk 4 above)
Comment (yimjay) [reply README.md]: Status updated

File: README.md (hunk 5 above)
Comment (yimjay) [reply README.md]: `research/abstractionAnalysis/design` will not be included in public release repo

File: README.md (hunk 6 above)
Comment (yimjay) [reply README.md:19]: Added the step: "Place the downloaded `POET` directory in the same directory as `abstraction-analysis`"

File: README.md (hunk 3 above)
Comment (yimjay) [reply README.md:1]: The Owners header is not included in my original README file. Would you like me to add a new section? Github automatically lists owners and contributors on the right column of a repo (at least on my end)

File: .gitattributes (hunk 7 above)
Comment (bnestere) [reply .gitattributes]: Hmm I think future proofing is probably a good idea. I found a repository with a bunch of sample `.gitattributes` files for various projects. Perhaps we should include the contents of the `Common` and `C++` ones.

**Review by qingyi-yan — APPROVED** [review]

File: .gitattributes (hunk 7 above)
Comment (yimjay) [reply .gitattributes]: Added `Common` and `C++` gitattributes :)

**Review by bnestere — APPROVED** [review]
Looks good to me. Thanks for the extension!

**Pull request merged**